src/main/java/io/github/adorjan/loader/PropertiesLoader.java

```diff
@@ -0,0 +1,40 @@
+package io.github.adorjan.loader;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.github.adorjan.Loader;
+
+public class PropertiesLoader implements Loader<Properties, Optional<InputStream>> {
+
+    private Logger logger = LoggerFactory.getLogger(PropertiesLoader.class);
+
+    @Override
+    public Optional<Properties> load(Optional<InputStream> inputStream) {
```

Comment:
- inputStream should belong to the configuration of the class (constructor)
- at this point the configuration data (inputstream) should not be optional - without an input stream there is nothing left for the object to do - pls consider throwing an exception
- if we configure the class with a specific inputstream to work with, there is no point to pass an argument to load
- it is still valid to return an optional of properties from load, as it is not guaranteed that the contents of the input stream can be parsed into a properties object